<div>
    <h1>ADMIN PANEL: USUARIOS</h1>
    {{#each users}}
        <div style="display: inline-block">
            <p id="{{this._id}}" class="userId">Nombre: {{this.first_name}} email: {{this.email}} Rol: {{this.role}}</p>
        </div>
        <div style="display: inline-block">
            <button data-userid="{{this._id}}" class="updateRole" onclick="updateRole(this)">Cambiar Rol</button>
            <button data-userid="{{this._id}}" class="deleteUser" onclick="deleteUser(this)">Eliminar usuario</button>
        </div>
            </br>
            </br>
    {{/each}}
        <button id="deleteInactiveUsers" class="deleteInactiveUsers" onclick="deleteInactiveUsers()">Delete inactive users</button>
    </br>
    </br>
    </br>
    </br>
    <a href='/auth/administracion'>Volver</a>
</div>


<script>
    

function updateRole(button) {

    const userId = button.getAttribute('data-userId');

    fetch(`/api/users/premium/${userId}`, {
        method: 'PUT',
    })
    .then(response => {
        if (response.ok) {
            alert('USER ROLE UPDATED');
            location.reload()
        } else {
            throw new Error('Failed to delete product from cart');
        }
    })
    .catch(error => {
        console.error(error);
    });
};

function deleteUser(button) {
    const userId = button.getAttribute('data-userId');
    
    fetch(`/api/users/${userId}`, {
        method: 'DELETE',
    })
    .then(response => {
        if (response.status === 200) {
            alert('USER DELETED');
            window.location.reload();
        } else {
            console.error('Error en la solicitud Fetch');
        }
    })
    .catch(error => {
        console.error('Error de red:', error);
    });
};


function deleteInactiveUsers() {

    fetch(`/api/users`, {
        method: 'DELETE',
    })
    .then(response => {
        if (response.status === 200) {
        alert('USUARIOS ELIMINADOS')
    } else {
        console.error('Error en la solicitud Fetch');
    }
    window.location.reload();
})
    .catch(error => {
        // Manejar errores de red aqu√≠
        console.error('Error de red:', error);
    });
};
</script>